### PR TITLE
Fix data race in eestream

### PR DESCRIPTION
**Pull request by kaloyan-raev**
Fixes https://storjlabs.atlassian.net/browse/V3-332

### CODE REVIEW

File: pkg/eestream/encode.go [hunk 1]
```diff
@@ -202,7 +204,8 @@ func (er *encodedReader) fillBuffer() {
 func (er *encodedReader) copyData(num int, copier <-chan block) {
 	// close the respective buffer channel when this goroutine exits
 	defer func() {
-		if er.eps[num].ch != nil {
+		if atomic.LoadInt32(er.eps[num].closed) == 0 {
+			atomic.StoreInt32(er.eps[num].closed, 1)
```
Comment (egonelbre) [inline pkg/eestream/encode.go]: This still has a race, at this point the closed status may already have changed.

Comment (egonelbre) [reply pkg/eestream/encode.go]:
``` 
// goroutine 1                                      // goroutine 2

if atomic.LoadInt32(er.eps[num].closed) == 0 {      
                                                    if atomic.LoadInt32(er.eps[num].closed) == 0 {
                                                        atomic.StoreInt32(er.eps[num].closed, 1)
    atomic.StoreInt32(er.eps[num].closed, 1)            close(er.eps[num].ch)
    close(er.eps[num].ch)
```

Comment (kaloyan-raev) [reply pkg/eestream/encode.go]: There is only one call to `copyData` per `num`.

Comment (egonelbre) [reply pkg/eestream/encode.go]: The other place has the exact same problem, so it has still the possibility for double close.

Comment (egonelbre) [reply pkg/eestream/encode.go]: I'll look again tomorrow with fresh eyes, maybe you are right.

Comment (kaloyan-raev) [reply pkg/eestream/encode.go]: But let me try using `CompareAndSwapInt32` instead.

**Pull request merged**